src/NCronJob/Registry/IInstantJobRegistry.cs

```diff
@@ -250,7 +249,17 @@ private Guid RunJob<TJob>(DateTimeOffset startDate, object? parameter = null, bo
     {
         using (logger.BeginScope("Triggering RunScheduledJob:"))
         {
-            var jobDefinition = jobRegistry.FindFirstJobDefinition(typeof(TJob));
+            var jobDefinitions = jobRegistry.FindAllJobDefinition(typeof(TJob));
```

Comment:
On an unrelated note, the `JobRegistry` could also offer generic version of `Find...`.

I guess in the future we want to move those checks into `UseNCronJob`.

Or maybe even now and already highlight this in the documentation. Still, this sanity check makes sense, but it would be nice to have it as early as possible

Reply:
>I guess in the future we want to move those checks into UseNCronJob.

That's the intent. UseNCronJob will be slowly enhanced to be able to detect weird registrations.